src/prof.c

```diff
@@ -799,42 +869,11 @@ prof_thread_name_get(tsd_t *tsd) {
 
 int
 prof_thread_name_set(tsd_t *tsd, const char *thread_name) {
-	assert(tsd_reentrancy_level_get(tsd) == 0);
-
-	prof_tdata_t *tdata;
-	unsigned i;
-	char *s;
-
-	tdata = prof_tdata_get(tsd, true);
-	if (tdata == NULL) {
-		return EAGAIN;
-	}
-
-	/* Validate input. */
-	if (thread_name == NULL) {
+	if (opt_prof_experimental_use_sys_thread_name) {
 		return EFAULT;
```

Comment:
EFAULT is probably the wrong thing to return here now. Maybe EINVAL or ENOENT, depending on the degree to which you regard the "use thread name" bool as an implicit argument to the function, or something that removes this mallctl from existence.

Reply:
`ENOENT` feels better to me.